src/t8_cmesh/t8_cmesh_netcdf.c

```diff
@@ -483,21 +484,56 @@ t8_cmesh_write_user_netcdf_vars (t8_cmesh_netcdf_context_t * context,
                                  t8_cmesh_netcdf_ugrid_namespace_t *
                                  namespace_context,
                                  int num_extern_netcdf_vars,
-                                 t8_netcdf_variable_t * ext_variables[])
+                                 t8_netcdf_variable_t * ext_variables[],
+                                 sc_MPI_Comm comm)
 {
 #if T8_WITH_NETCDF
-  int                 retval;
-  /* Check wheter user-defined variables should be written */
+  /* Check whether user-defined variables should be written */
   if (num_extern_netcdf_vars > 0 && ext_variables != NULL) {
-    for (int i = 0; i < num_extern_netcdf_vars; i++) {
+    int                 retval, i, int_mode;
+    int                 mpirank, mpisize;
+
+    retval = sc_MPI_Comm_size (comm, &mpisize);
+    SC_CHECK_MPI (retval);
+    retval = sc_MPI_Comm_rank (comm, &mpirank);
+    SC_CHECK_MPI (retval);
+
+    /* Iterate over the amount of user-defined variables */
+    for (i = 0; i < num_extern_netcdf_vars; i++) {
       switch (ext_variables[i]->datatype) {
       case 0:
         /* A NetCDF Integer-Variable will be declared */
-        if ((retval =
-             nc_def_var (context->ncid, ext_variables[i]->variable_name,
-                         NC_INT, 1, &context->nMesh_elem_dimid,
-                         &(ext_variables[i]->var_user_dimid)))) {
-          ERR (retval);
+        /* Check whether the data is corresponding to either a NC_INT variable (32bit) or a NC_INT64 variable (64bit) */
+        if (mpirank == 0) {
+          int_mode = (int) ext_variables[i]->var_user_data->elem_size;
+        }
+        retval = sc_MPI_Bcast (&int_mode, 1, sc_MPI_INT, 0, comm);
+        SC_CHECK_MPI (retval);
+        T8_ASSERT (int_mode ==
+                   (int) ext_variables[i]->var_user_data->elem_size);
+
+        if (int_mode == 4) {
```

Comment:
We should never hardcode such numbers as 4 and 8 for the byte sizes.
Use 'sizeof(int32_t)' etc.
This whole if construct may be obsolete with a redesign of the NC_INT macros.